File: src/main/java/de/thatsich/intellie/applied/aerodynamics/functionality/suite/chest/KeyHandler.java
package de.thatsich.intellie.applied.aerodynamics.functionality.suite.chest;

import cpw.mods.fml.client.registry.ClientRegistry;
import cpw.mods.fml.common.eventhandler.SubscribeEvent;
import cpw.mods.fml.common.gameevent.InputEvent;
import net.minecraft.client.settings.KeyBinding;
import org.lwjgl.input.Keyboard;

/**
 * @author thatsIch
 * @date 26.03.2014.
 */
public class KeyHandler {
	private static final int FLY_KEY = Keyboard.KEY_G;
	private static final KeyBinding BINDING = new KeyBinding("description", KeyHandler.FLY_KEY, "FLY_KEY.category");

	public KeyHandler() {
		ClientRegistry.registerKeyBinding(KeyHandler.BINDING);
	}

	@SubscribeEvent
	public void onKeyInput(InputEvent.KeyInputEvent event) {
		if (KeyHandler.BINDING.isPressed()) {
			System.out.println("PRESSED");
			//			AppliedAerodynamics.packetPipeline.sendToServer( new FlyPacket() );
		}
	}
}

File: src/main/scala/de/thatsich/minecraft/intellie/applied/aerodynamics/functional/suite/chest/KeyHandler.scala
package de.thatsich.minecraft.intellie.applied.aerodynamics.functional.suite.chest

import org.lwjgl.input.Keyboard
import net.minecraft.client.settings.KeyBinding
import cpw.mods.fml.client.registry.ClientRegistry
import cpw.mods.fml.common.eventhandler.SubscribeEvent
import cpw.mods.fml.common.gameevent.InputEvent

object KeyHandler
{
	private final val FLY_KEY: Int        = Keyboard.KEY_G
	private final val BINDING: KeyBinding = new KeyBinding("description", KeyHandler.FLY_KEY, "FLY_KEY.category")
}

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
class KeyHandler
{

	ClientRegistry.registerKeyBinding(KeyHandler.BINDING)

	@SubscribeEvent def onKeyInput(event: InputEvent.KeyInputEvent)
	{
		if( KeyHandler.BINDING.isPressed )
		{
			System.out.println("PRESSED")
		}
	}
}
